### PR TITLE
exam 1

### PR DESCRIPTION
**Pull request by anahittum**

### CODE REVIEW

File: src/com/aca/exam/exam1/Exam.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package com.aca.exam.exam1;
+
+import java.util.Arrays;
+import java.util.Scanner;
+
+public class Exam {
+
+    Scanner scanner = new Scanner(System.in);
```
Comment (gevorgyanarman) [inline src/com/aca/exam/exam1/Exam.java:8]: this can be a private

File: src/com/aca/exam/exam1/Exam.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    public int[] doubleValue() {
+        for (int i = 0; i < array.length; i++) {
+            array[i] = 2 * array[i];
+        }
+        return array;
+    }
+
+    public int average() {
+        int sum = 0;
+        int average;
```
Comment (gevorgyanarman) [inline src/com/aca/exam/exam1/Exam.java:50]: why is this declared here? the variable declaration should be as close to its first usage as possible. you can even not use it

File: src/com/aca/exam/exam1/Exam.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        for (int i = 0; i < array.length; i++) {
+            sum += array[i];
+        }
+        average = sum / array.length;
+        return average;
+    }
+
+    public int printNthNumber(int n) {
+        if (n < 0 || n >= array.length) {
+            return 0;
```
Comment (gevorgyanarman) [inline src/com/aca/exam/exam1/Exam.java:60]: What if array contains 0 value(s)

File: src/com/aca/exam/exam1/Exam.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        String convertedArray = Arrays.toString(array);
+        for (int i = 0; i < convertedArray.length(); i++) {
+            if (convertedArray.charAt(i) == '0') {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    public void endGame() {
```
Comment (gevorgyanarman) [inline src/com/aca/exam/exam1/Exam.java:81]: the existence of this method is not clear to me

File: src/com/aca/exam/exam1/ExamTest.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package com.aca.exam.exam1;
+
+import java.util.Arrays;
+
+public class ExamTest {
+    public static void main(String[] args) {
+        Exam exam = new Exam();
+        System.out.println(Arrays.toString(exam.getInputArray()));
+        System.out.println(exam.max());
```
Comment (gevorgyanarman) [inline src/com/aca/exam/exam1/ExamTest.java:9]: where is the function selection method? why you hardcoded the method invocations?

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

File: src/com/aca/exam/exam1/Exam.java (hunk 4 above)
Comment (anahittum) [reply src/com/aca/exam/exam1/Exam.java:81]: The method asks the user to input a value and the program is not terminated until the user input is 8.

File: src/com/aca/exam/exam1/Exam.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+    private final int length = 5;
+
+    private final int[] array = new int[length];
+
+    public Exam() {
+        getInputArray();
+    }
+
+    public int[] getInputArray() {
```
Comment (gevorgyanarman) [inline src/com/aca/exam/exam1/Exam.java:18]: if the method didn't change the global array variable but instead create a new one, the method return type would be justified. so either make it void or create a new array and return it and assign global array to the one returned by this method

File: src/com/aca/exam/exam1/ExamTest.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package com.aca.exam.exam1;
+
+public class ExamTest {
+    public static void main(String[] args) {
+        Exam exam = new Exam();
+        exam.methodSelector();
```
Comment (gevorgyanarman) [inline src/com/aca/exam/exam1/ExamTest.java:6]: here you have a single game. please make it repeatable until the user writes 8 number

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

File: src/com/aca/exam/exam1/Exam.java [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+                    System.out.println(printNthNumber());
+                    break;
+                case 6:
+                    System.out.println(minMaxDifference());
+                    break;
+                case 7:
+                    System.out.println(countOfZeros());
+                    break;
+                case 8:
+                    System.exit(0);
```
Comment (gevorgyanarman) [inline src/com/aca/exam/exam1/Exam.java:111]: We use System.exit in extreme situations. imagine a case where you should do some other processes when this method relation loop ends. please try to find a better alterative to stop the while loop

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

File: src/com/aca/exam/exam1/Exam.java [hunk 9]
```diff
@@ -108,8 +108,7 @@ public void methodSelector() {
                     System.out.println(countOfZeros());
                     break;
                 case 8:
-                    System.exit(0);
-                    break;
+                    return;
```
Comment (anahittum) [inline src/com/aca/exam/exam1/Exam.java:111]: @gevorgyanarman how about this?

**Pull request merged**